addons/common/fnc_dropMagazine.sqf

```diff
@@ -2,76 +2,56 @@
 Function: CBA_fnc_dropMagazine
 
 Description:
-    Drop a magazine.
+    Drops a magazine.
 
     Function which verifies existence of _item and _unit, returns false in case
     of trouble, or when able to remove _item from _unit true in case of success.
 
 Parameters:
-    _unit   - the unit that should drop a magazine [Object]
-    _item   - class name of the magazine to drop [String]
+    _unit - the unit that should drop a magazine <OBJECT>
+    _item - class name of the magazine to drop <STRING>
+    _ammo - ammo count <NUMBER>
 
 Returns:
-    true if successful, false otherwise
+    true if successful, false otherwise <BOOLEAN>
 
 Examples:
     (begin example)
-    _result = [player, "SmokeShell"] call CBA_fnc_dropMagazine
+        _result = [player, "SmokeShell"] call CBA_fnc_dropMagazine
     (end)
 
 Author:
-
+    ?, commy2
 ---------------------------------------------------------------------------- */
-
 #include "script_component.hpp"
 SCRIPT(dropMagazine);
 
-#define __scriptname fnc_dropMagazine
-
-#define __cfg (configFile >> "CfgMagazines")
-#define __action "DROPMAGAZINE"
-#define __ar (magazines _unit)
+params [["_unit", objNull, [objNull]], ["_item", "", [""]], ["_ammo", -1, [0]]];
 
-private ["_item", "_holder"];
-params ["_unit"];
-if (typeName _unit != "OBJECT") exitWith {
-    TRACE_2("Unit not Object",_unit,_item);
-    false
-};
-_item = _this select 1;
-if (typeName _item != "STRING") exitWith {
-    TRACE_2("Item not String",_unit,_item);
-    false
-};
-if (isNull _unit) exitWith {
-    TRACE_2("Unit isNull",_unit,_item);
-    false
-};
-if (_item == "") exitWith {
-    TRACE_2("Empty Item",_unit,_item);
-    false
-};
-if !(isClass (__cfg >> _item)) exitWith {
-    TRACE_2("Item not exist in Config",_unit,_item);
-    false
+// random mag mode
+if (_ammo < 0) then {

```

Comment:
done. also changed `selectRandom` to `BIS_fnc_selectRandom` for the Linux build.